browser/gui/app.cpp

```diff
@@ -470,9 +473,41 @@ void App::run_layout_widget() const {
 void App::clear_render_surface() {
     if (render_debug_) {
         window_.clear();
-    } else {
+        return;
+    }
+
+    if (!page_loaded_) {
         window_.clear(sf::Color(255, 255, 255));
+        return;
     }
+
+    // https://www.w3.org/TR/css-backgrounds-3/#special-backgrounds
+    // If html or body has a background set, use that as the canvas background.
+    // TODO(robinlinden): This should be done in //render, but requires new
+    //                    //gfx APIs that I want to think a bit about.
+    if (auto html_bg = engine_.layout().get_property<css::PropertyId::BackgroundColor>();
+            html_bg != gfx::Color::from_css_name("transparent")) {
+        window_.clear(sf::Color(html_bg->as_rgba_u32()));
+        return;
+    }
+
+    auto body = std::ranges::find_if(engine_.layout().children, [](layout::LayoutBox const &c) {
```

Comment:
isn't it time to write a tree selector api?

Reply:
I have written `dom::nodes_by_xpath`, and I have extended it to work with both the style and layout tree, but all of that stuff is blocked waiting for reviews. Any improvements to that function will automagically work for the trees wrapping dom nodes once I've finished merging the changes to it.

TL;DR: it was time a while ago, and it is done*. :P